engine/src/test/java/io/camunda/zeebe/engine/state/instance/JobStateTest.java

```diff
@@ -646,6 +646,7 @@ public void testInvariants() {
         .hasStackTraceContaining("type must not be empty");
 
     // cancel
+    jobState.create(1L, newJobRecord());
```

Comment:
:question: Please explain the change in this test. Is it because activate/complete now checks preconditions?

Reply:
It's necessary so that the following cancel doesn't fail too early. In the next line we check that the cancelling of a job without type fails with a stacktrace that contains "type must not be empty". Without first creating a valid job with the same key, the cancelling instead fails earlier, because it tries to `deleteExisting` a job that doesn't exist.